ast/testcase/testcase.go

```diff
@@ -105,137 +101,117 @@ func Transform(ctx context.Context, pkg *ast.Package, modules TestModules) ([]st
 	return names, pkgs, nil
 }
 
-func newTestPackage(ctx context.Context, basePkg *ast.Package, predicate []ast.Statement, tc *ast.TestCaseStatement, modules TestModules) (*ast.Package, error) {
+func newTestPackage(ctx context.Context, basePkg *ast.Package, preamble []ast.Statement, tc *ast.TestCaseStatement, modules TestModules) (*ast.Package, error) {
 	pkg := basePkg.Copy().(*ast.Package)
 	pkg.Package = "main"
 	pkg.Files = nil
 
 	file := basePkg.Files[0].Copy().(*ast.File)
 	file.Package.Name.Name = "main"
 
-	file.Body = make([]ast.Statement, 0, len(predicate)+len(tc.Block.Body))
-	file.Body = append(file.Body, predicate...)
+	file.Body = make([]ast.Statement, 0, len(preamble)+len(tc.Block.Body))
+	file.Body = append(file.Body, preamble...)
 	if tc.Extends != nil {
-		f, err := extendTest(file, tc.Extends.Value, modules)
+		parentImports, parentPreamble, parentTC, err := extendTest(file, tc.Extends.Value, modules)
 		if err != nil {
 			return nil, err
 		}
-		pkg.Files = append(pkg.Files, f)
+		file.Imports = mergeImports(file.Imports, parentImports)
+		file.Body = append(file.Body, parentPreamble...)
+		// Copy test case statements into body replacing the testing.do statement
+		// with the parent test case statements
+		found := false
+		for _, s := range tc.Block.Body {
+			if !found {
+				if es, ok := s.(*ast.ExpressionStatement); ok {
+					if call, ok := es.Expression.(*ast.CallExpression); ok {
+						if member, ok := call.Callee.(*ast.MemberExpression); ok && member.Property.Key() == "do" {
```

Comment:
To simplify this, since this is a find/replace, I would just use `super()` as a raw name. It's common for `super()` to have a special meaning in a programming language and specifying `testing.do()` makes me want to try to find the function. Since the function doesn't exist, I think that's confusing.